bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1023,4 +1023,82 @@ impl<'a> CircuitInputStateRef<'a> {
         }
         Ok(())
     }
+
+    pub(crate) fn gen_restore_context_ops(
+        &mut self,
+        exec_step: &mut ExecStep,
+        geth_steps: &[GethExecStep],
+    ) -> Result<(), Error> {
+        let geth_step = &geth_steps[0];
+        let call = self.call()?.clone();
+        if !call.is_success {
+            // add call failure ops for exception cases
+            self.call_context_read(
+                exec_step,
+                call.call_id,
+                CallContextField::IsSuccess,
+                0u64.into(),
+            );
+            self.call_context_read(
+                exec_step,
+                call.call_id,
+                CallContextField::IsPersistent,
+                0u64.into(),
+            );
+            if call.is_root {
+                return Ok(());
+            }
+        }
+
+        let caller = self.caller()?.clone();
+        self.call_context_read(
+            exec_step,
+            call.call_id,
+            CallContextField::CallerId,
+            caller.call_id.into(),
+        );
+
+        let geth_step_next = &geth_steps[1];
+        let caller_ctx = self.caller_ctx()?;
+        let caller_gas_left = if geth_step_next.gas.0 == 0 {
+            0
+        } else {
+            geth_step_next.gas.0 - geth_step.gas.0
+        };
```

Comment:
This should be:
```suggestion
        let caller_gas_left = if call.is_success {
            geth_step_next.gas.0 - geth_step.gas.0
        } else {
            geth_step_next.gas.0
        };
```

Reply:
thanks for catching , fixed in [5a82b5d](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/746/commits/5a82b5dfb8283bebd0d4f869078104d7e6d35252) thanks!